### PR TITLE
Add MESA_framebuffer_flip_y extension

### PR DESCRIPTION
**Pull request by fritzk**
Adds an extension to glFramebufferParameteri
that will specify if the framebuffer is vertically
flipped. Historically system framebuffers are
vertically flipped and user framebuffers are not.
Checking to see the state was done by looking at
the name field.  This adds an explicit field.

### CODE REVIEW

**Comment by fritzk** [comment]
This extension has been moving through the mesa list.  (https://lists.freedesktop.org/archives/mesa-dev/2018-July/199946.html)

It is at the point where there isn't any objection to the idea and I would like to begin formalizing it.

**Comment by chadversary** [comment]
+1

File: extensions/MESA/MESA_framebuffer_flip_y.txt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    Accepted by the <pname> argument of FramebufferParameteri and
+    GetFramebufferParameteriv:
+
+        GL_FRAMEBUFFER_FLIP_Y_MESA                      0x8BBB
+
+Errors
+
+    An INVALID_OPERATION error is generated by GetFramebufferParameteriv if
+    the default framebuffer is bound to and is FRAMEBUFFER_FLIP_Y_MESA.
```
Comment (chadversary) [inline extensions/MESA/MESA_framebuffer_flip_y.txt]: Oops. Github's Markdown mangled my earlier reply. This block should read:

`An INVALID_OPERATION error is generated by GetFramebufferParameteriv if
the default framebuffer is bound to <target> and <pname> is FRAMEBUFFER_FLIP_Y_MESA.`

**Comment by oddhack** [comment]
From a mechanics view, this is fine - just a question of when you want it merged. Is @chadversary the only person you want signing off on the spec aspects before doing that?

File: extensions/MESA/MESA_framebuffer_flip_y.txt (hunk 1 above)
Comment (fritzk) [reply extensions/MESA/MESA_framebuffer_flip_y.txt]: thanks.  i thought it sounded odd, should have spoken up.

**Comment by fritzk** [comment]
@chadversary is the only one needed for sign-ff

**Pull request merged**